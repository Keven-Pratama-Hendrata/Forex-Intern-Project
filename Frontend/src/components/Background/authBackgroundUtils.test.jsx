import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from '@jest/globals';

import {
  getCircleConfigs,
  getConfig,
  renderWaveBackground,
  renderCircles,
} from './authBackgroundUtils';

describe('authBackgroundUtils – config helpers', () => {
  it('returns fresh, complete circle-config objects', () => {
    const cfg1 = getCircleConfigs();
    const cfg2 = getCircleConfigs();

    expect(Object.keys(cfg1)).toEqual(['login', 'signup']);
    expect(cfg1.login).toEqual(cfg2.login);
    expect(cfg1.signup).toEqual(cfg2.signup);

    expect(cfg1).not.toBe(cfg2);
    expect(cfg1.login).not.toBe(cfg2.login);
  });

  it('getConfig selects explicit position or falls back to login', () => {
    const cfg = getCircleConfigs();
    expect(getConfig('login', cfg)).toBe(cfg.login);
    expect(getConfig('signup', cfg)).toBe(cfg.signup);
    expect(getConfig('does-not-exist', cfg)).toBe(cfg.login);
    expect(getConfig(undefined, cfg)).toBe(cfg.login);
  });
});

describe('authBackgroundUtils – SVG render helpers', () => {
  it('renderWaveBackground outputs a full-screen gradient SVG', () => {
    const { container } = render(renderWaveBackground());
    const svg = container.querySelector('svg');

    expect(svg).toHaveAttribute('viewBox', '0 0 100 100');
    expect(svg).toHaveClass(
      'absolute',
      'inset-0',
      'h-full',
      'w-full',
      'pointer-events-none',
    );

    expect(container.querySelector('linearGradient')).toHaveAttribute('id', 'bgGrad');
    expect(container.querySelectorAll('stop')).toHaveLength(2);
    expect(container.querySelector('rect')).toHaveAttribute('fill', 'url(#bgGrad)');
    expect(container.querySelector('path')).toHaveAttribute('fill', '#ffffff');

    expect(svg).toMatchSnapshot();
  });

  it('renderCircles draws a custom-class SVG with circles', () => {
    const cfg = {
      className: 'test-class',
      circles: [
        { cx: 90, cy: 70, r: 55 },
        { cx: 130, cy: 60, r: 40 },
      ],
    };

    const { container } = render(renderCircles(cfg));
    const svg = container.querySelector('svg');

    expect(svg).toHaveClass('test-class');
    expect(svg).toHaveAttribute('viewBox', '0 0 200 200');
    expect(svg).toHaveAttribute('stroke-width', '3');

    const circles = container.querySelectorAll('circle');
    expect(circles).toHaveLength(2);
    expect([...circles].map((c) => c.getAttribute('r'))).toEqual(['55', '40']);

    const empty = render(renderCircles({ className: 'x', circles: [] }))
      .container.querySelectorAll('circle');
    expect(empty).toHaveLength(0);
  });
});
