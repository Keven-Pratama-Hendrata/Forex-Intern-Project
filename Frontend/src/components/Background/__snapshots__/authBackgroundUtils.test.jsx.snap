// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`authBackgroundUtils â€“ SVG render helpers renderWaveBackground outputs a full-screen gradient SVG 1`] = `
<svg
  class="absolute inset-0 h-full w-full pointer-events-none"
  preserveAspectRatio="none"
  viewBox="0 0 100 100"
>
  <defs>
    <lineargradient
      id="bgGrad"
      x1="0"
      x2="0"
      y1="0"
      y2="1"
    >
      <stop
        offset="0%"
        stop-color="#6ca0c1"
      />
      <stop
        offset="100%"
        stop-color="#507fa1"
      />
    </lineargradient>
  </defs>
  <rect
    fill="url(#bgGrad)"
    height="100"
    width="100"
  />
  <path
    d="M 70 0 Q 77 10 70 20 Q 63 30 70 40 Q 77 50 70 60 Q 63 70 70 80 Q 77 90 70 100 L 100 100 100 0 Z"
    fill="#ffffff"
  />
</svg>
`;
